src/network/src/esp/src/TCPClient.cpp

```diff
@@ -0,0 +1,58 @@
+#include "TCPClient.h"
+#include "NetworkConfig.h"
+#include "SocketFactory.h"
+#include "lwip/sockets.h"
+
+TCPClient::TCPClient(ILogger& logger) : m_logger(logger) {
+    m_hasSocket = false;
+    m_socketFd = NO_SOCKET;
```

Comment:
You could do like in the HiveMind and instead of having a bool + a value, using an std::optional for `m_socketFd`

Reply:
I'll just remove the bool and check the value of the socket. optional is overkill for an int.